src/mscorlib/src/System/TimeZoneInfo.Unix.cs

```diff
@@ -381,26 +381,34 @@ private static bool TryLoadTzFile(string tzFilePath, ref byte[] rawData, ref str
         /// </summary>
         private static string FindTimeZoneIdUsingReadLink(string tzFilePath)
         {
-            string id = null;
-
-            StringBuilder symlinkPathBuilder = StringBuilderCache.Acquire(Path.MaxPath);
-            bool result = Interop.GlobalizationInterop.ReadLink(tzFilePath, symlinkPathBuilder, (uint)symlinkPathBuilder.Capacity);
-            if (result)
+            int capacity = 4096;
+            do
             {
-                string symlinkPath = StringBuilderCache.GetStringAndRelease(symlinkPathBuilder);
-                // time zone Ids have to point under the time zone directory
-                string timeZoneDirectory = GetTimeZoneDirectory();
-                if (symlinkPath.StartsWith(timeZoneDirectory))
+                StringBuilder symlinkPathBuilder = StringBuilderCache.Acquire(capacity);
```

Comment:
StringBuilderCache caches up to MAX_BUILDER_SIZE = 360 only. You are starting with 4k here, so StringBuilderCache is not going do anything here.

Reply:
We can start at a lower value. `4096` is what was used in practice before, so I started at that.